### ISSUE TITLE
auxiliary/scanner/http/wordpress_login_enum does not correctly enumerate logins on WordPress 5.6.1

### ISSUE THREAD

**Issue by gwillcox-r7** [rapid7/metasploit-framework#14790]
## Steps to reproduce

How'd you do it?

1. `mkdir wordpress`
2. `cd wordpress`
3. `nano docker-compose.yml`
4. Enter the following into the file
```
version: '3.3'

services:
   db:
     image: mysql:5.7
     volumes:
       - db_data:/var/lib/mysql
     restart: always
     environment:
       MYSQL_ROOT_PASSWORD: somewordpress
       MYSQL_DATABASE: wordpress
       MYSQL_USER: wordpress
       MYSQL_PASSWORD: wordpress

   wordpress:
     depends_on:
       - db
     image: wordpress:latest
     ports:
       - "8000:80"
     restart: always
     environment:
       WORDPRESS_DB_HOST: db:3306
       WORDPRESS_DB_USER: wordpress
       WORDPRESS_DB_PASSWORD: wordpress
       WORDPRESS_DB_NAME: wordpress
volumes:
    db_data: {}
```
5. `sudo docker-compose up -d`
6. Follow the setup instructions at `127.0.0.1:8000` and set up a new user in WordPress.
7. Start `msfconsole`
8. `creds add user:*username from setup*`
9. `use auxiliary/scanner/http/wordpress_login_enum`
10. `set RHOSTS 127.0.0.1`
11. `set RPORT 8000`
12. `set DB_ALL_USERS true`
13. `run`

## Were you following a specific guide/tutorial or reading documentation?

Was testing https://github.com/rapid7/metasploit-framework/pull/14748 when I discovered this issue.

## Expected behavior

The module should correctly identify that the user that was created during the setup instructions exists in the system.

## Current behavior

The module does not correctly identify that users exist on the target WordPress installation, even if they are valid.

```
msf6 > use auxiliary/scanner/http/wordpress_login_enum 
msf6 auxiliary(scanner/http/wordpress_login_enum) > show options

Module options (auxiliary/scanner/http/wordpress_login_enum):

   Name                 Current Setting  Required  Description
   ----                 ---------------  --------  -----------
   BLANK_PASSWORDS      false            no        Try blank passwords for all users
   BRUTEFORCE           true             yes       Perform brute force authentication
   BRUTEFORCE_SPEED     5                yes       How fast to bruteforce, from 0 to 5
   DB_ALL_CREDS         false            no        Try each user/password couple stored in the current database
   DB_ALL_PASS          false            no        Add all passwords in the current database to the list
   DB_ALL_USERS         false            no        Add all users in the current database to the list
   ENUMERATE_USERNAMES  true             yes       Enumerate usernames
   PASSWORD                              no        A specific password to authenticate with
   PASS_FILE                             no        File containing passwords, one per line
   Proxies                               no        A proxy chain of format type:host:port[,type:host:port][...]
   RANGE_END            10               no        Last user id to enumerate
   RANGE_START          1                no        First user id to enumerate
   RHOSTS                                yes       The target host(s), range CIDR identifier, or hosts file with syntax 'file:<path>'
   RPORT                80               yes       The target port (TCP)
   SSL                  false            no        Negotiate SSL/TLS for outgoing connections
   STOP_ON_SUCCESS      false            yes       Stop guessing when a credential works for a host
   TARGETURI            /                yes       The base path to the wordpress application
   THREADS              1                yes       The number of concurrent threads (max one per host)
   USERNAME                              no        A specific username to authenticate as
   USERPASS_FILE                         no        File containing users and passwords separated by space, one pair per line
   USER_AS_PASS         false            no        Try the username as the password for all users
   USER_FILE                             no        File containing usernames, one per line
   VALIDATE_USERS       true             yes       Validate usernames
   VERBOSE              true             yes       Whether to print output for all attempts
   VHOST                                 no        HTTP server virtual host

msf6 auxiliary(scanner/http/wordpress_login_enum) > set RHOSTS 127.0.0.1
RHOSTS => 127.0.0.1
msf6 auxiliary(scanner/http/wordpress_login_enum) > set RPORT 8080
RPORT => 8080
msf6 auxiliary(scanner/http/wordpress_login_enum) > set DB_ALL_USERS true
DB_ALL_USERS => true
msf6 auxiliary(scanner/http/wordpress_login_enum) > run

[-] The connection was refused by the remote host (127.0.0.1:8080).
[-] / does not seem to be WordPress site
[*] Scanned 1 of 1 hosts (100% complete)
[*] Auxiliary module execution completed
msf6 auxiliary(scanner/http/wordpress_login_enum) > set RPORT 8000
RPORT => 8000
msf6 auxiliary(scanner/http/wordpress_login_enum) > run

[*] / - WordPress Version 5.6.1 detected
[*] 127.0.0.1:8000 - / - WordPress User-Enumeration - Running User Enumeration
[*] 127.0.0.1:8000 - / - WordPress User-Validation - Running User Validation
[*] / - WordPress User-Validation - Checking Username:'Administrator'
[-] 127.0.0.1:8000 - [1/4] - / - WordPress User-Validation - Invalid Username: 'Administrator'
[*] / - WordPress User-Validation - Checking Username:'administrator'
[-] 127.0.0.1:8000 - [2/4] - / - WordPress User-Validation - Invalid Username: 'administrator'
[*] / - WordPress User-Validation - Checking Username:'test'
[-] 127.0.0.1:8000 - [3/4] - / - WordPress User-Validation - Invalid Username: 'test'
[*] / - WordPress User-Validation - Checking Username:'normal'
[-] 127.0.0.1:8000 - [4/4] - / - WordPress User-Validation - Invalid Username: 'normal'
[*] 127.0.0.1:8000 - [5/4] - / - WordPress Brute Force - Running Bruteforce
[*] Scanned 1 of 1 hosts (100% complete)
[*] Auxiliary module execution completed
msf6 auxiliary(scanner/http/wordpress_login_enum) > 
```

Here the username `test` should be a valid user, but it is not identified as one.

### Metasploit version

6.0.32-dev-5c886d46bf

**Comment by adfoster-r7** [comment]
We have a ticket to look at updating the wordpress mixins to work with 5.x that might overlap with this

**Comment by dthakkar845** [comment]
I noticed that when wordpress site is trying to confirm or verify your email address as soon as you login, the auxiliary/scanner/http/wordpress_login_enum module fails to validate correct userid and password. The script must be looking for some banner on wpsite to confirm successful login. When I confirm my email address on the site after manually loging in and then try metasploit module subsequently, it correctly finds id and password.

**Comment by bcoles** [comment]
> I noticed that when wordpress site is trying to confirm or verify your email address as soon as you login, the auxiliary/scanner/http/wordpress_login_enum module fails to validate correct userid and password. The script must be looking for some banner on wpsite to confirm successful login. When I confirm my email address on the site after manually loging in and then try metasploit module subsequently, it correctly finds id and password.

If you run the module with `set HttpTrace true` you will receive more verbose output. You should then be able to compare the responses before and after verifying your email.

At a guess, the failing comparison might be in `wordpress_user_exists?(user)` here:

https://github.com/rapid7/metasploit-framework/blob/aa2a8433fe8d2ccaa52a42dc63b50beeb66dd7bf/lib/msf/core/exploit/remote/http/wordpress/users.rb#L4-L21

**Comment by dthakkar845** [comment]
I used auxiliary scanner with username set to ‘admin’ I have password file
with correct password “qwerty123” for admin and when I run the module it
said password qwerty123 not valid for user ‘admin’. Puzzled, I logged into
wpsite  manually and there was a prompt asking me if my email is correct.
Once I said yes to that prompt I was on wpsite logged in. After that I
logged out of wpsite and tried auxiliary scanner again and this time it
said password qwerty123 was valid for user admin.
